packages/react-router-dom/modules/NavLink.js

```diff
@@ -38,8 +38,9 @@ function NavLink({
         );
         const { pathname: path } = toLocation;
         // Regex taken from: https://github.com/pillarjs/path-to-regexp/blob/master/index.js#L202
+        // but without escaping of slashes and dots
         const escapedPath =
-          path && path.replace(/([.+*?=^!:${}()[\]|/\\])/g, "\\$1");
+          path && path.replace(/([+*?=^!:${}()[\]|\\])/g, "\\$1");
```

Comment:
This works because responding to the regex pattern below, slashes don't need to be escaped and dots need to be escaped by as backslash in the path:
`/(?:\\:(\\w+)(?:\\(((?:\\\\.|[^\\\\()])+)\\))?|\\(((?:\\\\.|[^\\\\()])+)\\))([+*?])?/`
(see: https://github.com/pillarjs/path-to-regexp/blob/d1ec03afb6eefe7ad19e4d990a428e6b545fb3a6/index.js#L29)